sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotOverwriteDictionaryValues.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    class TestCases
+    {
+        void Indexers(Dictionary<int, int> items)
+        {
+            int index;
+
+            items[index] = 1; // Secondary
+            items[index] = 2; // Noncompliant {{Verify this is the key that was intended; a value has already been set for it.}}
+//          ^^^^^^^^^^^^^^^^
+
+            items[0] = 1; // Secondary
+            items[1] = 2;
+            items[0] = 3; // Noncompliant
```

Comment:
How many errors would we expect to see raised if there is more than one duplicate for an index? 
```
items[0] = 1;
items[0] = 2;  // error here?
items[0] = 3; //... and error here?
```

Reply:
We will raise for each except the first. I suspect they did it because of the kill the nouse initiative, but in java they raise secondary locations for each of the subsequent assignments. Not sure if this is a good idea, because the secondary locations would show not only the first initialization, but subsequent overwrites too. I think this would be a rare case.